backend/filen/filen.go

```diff
@@ -0,0 +1,1084 @@
+
+Get this using the Filen CLI export-api-key command
+You can download the Filen CLI from https://github.com/FilenCloudDienste/filen-cli`,
+				Required:   true,
+				IsPassword: true,
+				Sensitive:  true,
+			},
+			{
+				Name:     config.ConfigEncoding,
+				Help:     config.ConfigEncodingHelp,
+				Advanced: true,
+				Default:  encoder.Standard | encoder.EncodeInvalidUtf8,
+			},
+			{
+				Name: "max_download_threads",
+				Help: `Max number of threads to use when downloading files.
+
+Each thread uses up to a bit over 1 MiB of memory.
+`,
+				Advanced: true,
+				Default:  sdk.DefaultMaxDownloadThreads,
+			},
+			{
+				Name: "max_download_threads_per_file",
+				Help: `Max number of threads per file to use when downloading files.
+
+Each thread uses up to a bit over 1 MiB of memory.
+`,
+				Advanced: true,
+				Default:  sdk.DefaultMaxDownloadThreadsPerFile,
+			},
+			{
+				Name: "max_upload_threads",
+				Help: `Max number of threads to use when uploading files.
+
+Each thread uses up to a bit over 1 MiB of memory.
+`,
+				Advanced: true,
+				Default:  sdk.DefaultMaxUploadThreads,
+			},
+			{
+				Name:      "master_keys",
+				Help:      "Master Keys (internal use only)",
+				Sensitive: true,
+				Advanced:  true,
+			}, {
+				Name:      "private_key",
+				Help:      "Private RSA Key (internal use only)",
+				Sensitive: true,
+				Advanced:  true,
+			}, {
+				Name:      "public_key",
+				Help:      "Public RSA Key (internal use only)",
+				Sensitive: true,
+				Advanced:  true,
+			}, {
+				Name:     "auth_version",
+				Help:     "Authentication Version (internal use only)",
+				Advanced: true,
+			}, {
+				Name:      "base_folder_uuid",
+				Help:      "UUID of Account Root Directory (internal use only)",
+				Sensitive: true,
+				Advanced:  true,
+			},
+		},
+	})
+}
+
+// NewFs constructs a Fs at the path root
+func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, error) {
+	opt := new(Options)
+	err := configstruct.Set(m, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	password, err := obscure.Reveal(opt.Password)
+	if err != nil {
+		return nil, fmt.Errorf("failed to reveal password: %w", err)
+	}
+	apiKey, err := obscure.Reveal(opt.APIKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to reveal api key: %w", err)
+	}
+
+	var filen *sdk.Filen
+	if password == "INTERNAL" {
+		tsconfig := sdk.TSConfig{
+			Email:          opt.Email,
+			MasterKeys:     strings.Split(opt.MasterKeys, "|"),
+			APIKey:         apiKey,
+			PublicKey:      opt.PublicKey,
+			PrivateKey:     opt.PrivateKey,
+			AuthVersion:    opt.AuthVersion,
+			BaseFolderUUID: opt.BaseFolderUUID,
+		}
+		filen, err = sdk.NewFromTSConfig(tsconfig)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		filen, err = sdk.NewWithAPIKey(ctx, opt.Email, password, apiKey)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	filen.MaxDownloadThreadsPerFile = opt.MaxDownloadThreadsPerFile
```

Comment:
Are these actually used in the backend's usage? Given that rclone backends stream everything and don't read or write to files directly?